### DISCUSSION TITLE
After performing splitByPlane, the topology of the mesh is changed

### DISCUSSION THREAD

**Discussion by huangdade** [elalish/manifold#1097] [Q&A]
origin mesh:
![QQ_1733274665675](https://github.com/user-attachments/assets/b31c088e-e7bf-49c6-87cc-5861ec181534)
after split:
![QQ_1733274672899](https://github.com/user-attachments/assets/47548d66-7b26-439f-99da-6563f40efa8b)

I want to keep the origin topology.

**Comment by pca006132** [comment]
It seems that the feature size is quite small, which causes some vertices to be removed and triangulated again. Which version are you using? I don't think with double precision, we should collapse those vertices...

**Reply by elalish** [reply]
Perhaps they are coplanar faces? If so and you need to keep us from collapsing them, you can ensure that by adding disjoint properties to those vertices.

**Reply by huangdade** [reply]
"version": "2.5.1", javascript

**Reply by huangdade** [reply]
yes, they are coplanar faces. I painted some faces into different colors.

**Reply by elalish** [reply]
Then just supply your colors as property channels and we should take care of it properly.

**Reply by elalish** [reply]
And try 3.0!

**Reply by huangdade** [reply]
But I supplied "runIndex" and "runOriginalID" to the Mesh. Isn't that enough?
![image](https://github.com/user-attachments/assets/8a80d5c3-f70b-48db-a14f-2d91ac4623df)

**Reply by huangdade** [reply]
I divided triangles into several groups and binded each group with one color.

**Reply by pca006132** [reply]
Please try the latest version first.

**Reply by elalish** [reply]
Hmm, yes it should be. It looks like you have the makings of a good test - want to minimize it and submit a PR to show it failing? You can add it right here: https://github.com/elalish/manifold/blob/master/bindings/wasm/examples/bindings.test.ts

**Reply by huangdade** [reply]
<img width="410" alt="image" src="https://github.com/user-attachments/assets/4f7c68ed-fcce-4d13-aa81-f01497eaef69">
<img width="361" alt="image" src="https://github.com/user-attachments/assets/9e47f5d9-2a0f-4327-ae57-0e8f7760d397">

I attached some properties to the vert. It still failed.

**Reply by elalish** [reply]
Would you mind submitting your failing test as a PR? That's much easier for us to work with than screenshots.

**Reply by huangdade** [reply]
ok.
